### ISSUE TITLE
first upload

### ISSUE THREAD

**Issue by lucis13** [lucis13/HPANTN#1]
import sys
import os
from keras.preprocessing.image import ImageDataGenerator
from keras import optimizers
from keras.models import Sequential
from keras.layers import Dropout, Flatten, Dense, Activation, GlobalMaxPooling2D
from keras.layers.convolutional import Conv2D, MaxPooling2D
from keras import callbacks
from keras.layers import LeakyReLU

train_data_path = './training_seg'
validation_data_path = './int_validation_seg'
ext_data_path = './ext_validation_seg'

"""
Parameters
"""
img_width, img_height = 250, 250
epochs = 100
batch_size = 64
classes_num = 9
lr = 0.0004

model = Sequential()
model.add(Conv2D(32, (3, 3), input_shape=(img_width, img_height, 3)))
model.add(Activation("relu"))
model.add(Conv2D(64, (3, 3)))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(3, 3)))

model.add(Conv2D(64, (5, 5)))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(3, 3)))

model.add(Conv2D(128, (3, 3)))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(3, 3)))

model.add(Conv2D(128, (3, 3)))
model.add(Activation("relu"))
model.add(MaxPooling2D(pool_size=(2, 2)))

model.add(Flatten())
model.add(Dense(1000))
model.add(Activation("relu"))
model.add(Dropout(0.5))
model.add(Dense(classes_num, activation='softmax'))

model.compile(loss='categorical_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])

train_datagen = ImageDataGenerator(
    rescale=1. / 255,
    shear_range=0.2,
    zoom_range=0.2,
    horizontal_flip=True)

test_datagen = ImageDataGenerator(rescale=1. / 255)

ext_datagen = ImageDataGenerator()

train_generator = train_datagen.flow_from_directory(
    train_data_path,
    target_size=(img_height, img_width),
    batch_size=batch_size,
    class_mode='categorical')

validation_generator = test_datagen.flow_from_directory(
    validation_data_path,
    target_size=(img_height, img_width), 
    batch_size=batch_size,
    class_mode='categorical')

ext_validation_generator = ext_datagen.flow_from_directory(
    ext_data_path,
    target_size=(img_height, img_width), 
    batch_size=batch_size,
    shuffle = False,
    class_mode=None)

"""
Tensorboard log
"""

log_dir = './tf-log/'
tb_cb = callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1)
cbks = [tb_cb]

model.load_weights('models/V_11th_weights.12-0.78.hdf5')

nb_train_samples = train_generator.samples
nb_validation_samples = validation_generator.samples

from keras.callbacks import EarlyStopping, ModelCheckpoint
callbacks = [
    EarlyStopping(monitor='val_loss', patience=10, verbose=0),
    ModelCheckpoint('./models/V_11th_weights.{epoch:02d}-{val_loss:.2f}.hdf5', monitor='val_loss', save_best_only=True, verbose=0),
]

hist = model.fit_generator(
    train_generator,
    steps_per_epoch=nb_train_samples // batch_size,
    epochs=epochs,
    validation_data=validation_generator,
    validation_steps=nb_validation_samples // batch_size,
    callbacks = callbacks)
print(hist.history)

y_predict = model.predict_generator(ext_validation_generator, verbose = 1)
import numpy
y_classes = y_predict.argmax(axis=-1)
label_map = (train_generator.class_indices)
label_map = dict((v,k) for k,v in label_map.items()) #flip k,v
numpy.savetxt("y_predict2.csv", y_predict, delimiter=",")
y_classes = y_predict.argmax(axis=-1)
y_classes = [label_map[k] for k in y_classes]
filelist = ext_validation_generator.filenames

import csv
csvfile = "predictlist.csv"
#Assuming res is a flat list
with open(csvfile, "w") as output:
    writer = csv.writer(output, lineterminator='\n')
    for val in filelist:
        writer.writerow([val])

target_dir = './models/'
if not os.path.exists(target_dir):
  os.mkdir(target_dir)
model.save('./models/5th_model.h5')
model.save_weights('./models/V_9th(resize)_weights.h5')

from os import listdir
from os.path import isfile, join
onlyfiles = [f for f in listdir("./ext_validation_seg/A") if isfile(join("./ext_validation_seg/A", f))]
filelist = numpy.asarray(onlyfiles)
numpy.savetxt("file_list", filelist, delimiter=",")
